### PR TITLE
6.0.1 - meta padding fixes

### PR DESCRIPTION
**Pull request by barncastle**
Tested against a parser, everything works now except CriteriaTree, my version has a WDBC signature but a WDB2 header - help!

Also ItemModifiedAppearance.dbd had an additional column compared to my version - can someone confirm the structure.

### CODE REVIEW

File: definitions/GroupFinderCategory.dbd [hunk 1]
```diff
@@ -1,15 +1,12 @@
 COLUMNS
-uint field_3184919908?
 locstring name_lang
-uint field_1299825360?
 int ID
 uint flags
 uint orderIndex
 
 BUILD 6.0.1.18179
-field_3184919908<32>
```
Comment (bloerwald) [inline definitions/GroupFinderCategory.dbd]: These defs come from DBMeta. Are you really sure they had the wrong column count?! Are you sure you have checked with 18179?

File: definitions/GarrFamilyName.dbd [hunk 2]
```diff
@@ -1,7 +1,7 @@
 COLUMNS
-uint field_2840433098?
 string field_803567487?
+int ID
```
Comment (bloerwald) [inline definitions/GarrFamilyName.dbd]: To be technically correct, this is a guessed name ;)

File: definitions/GarrPlotUICategory.dbd [hunk 3]
```diff
@@ -1,12 +1,11 @@
 COLUMNS
-uint field_329935761?
 locstring categoryName_lang
 uint field_2890979807?
 int ID
 uint plotType
 
 BUILD 6.0.1.18179
-field_329935761<32>
+$noninline,id$ID<32>
```
Comment (bloerwald) [inline definitions/GarrPlotUICategory.dbd]: This is the only noninline in this diff and version. Are you sure about that?

File: definitions/GroupFinderCategory.dbd (hunk 1 above)
Comment (barncastle) [reply definitions/GroupFinderCategory.dbd]: Yeah this is wrong. When I originally pulled the DBs from Martin's site any that failed I got the previous build.

Basically everything I've done is crap and needs redoing.

File: definitions/GarrPlotUICategory.dbd (hunk 3 above)
Comment (barncastle) [reply definitions/GarrPlotUICategory.dbd]: For the record, this isn't noninline, previous build issue again.

**Comment by Marlamin** [comment]
Dangit. I'll have to see if my extract is correct.

**Comment by Marlamin** [comment]
Looks like (in theory) my extract might be correct. 12GB zip of DBCs going from 6.0.1.18125 to 7.3.x [here](https://marlamin.com/bigdl/DBCs6xto7x.zip). Please double check if so, though. 8.0 isn't included as this is a pre-8.0 extract.

**Comment by barncastle** [comment]
@Marlamin I think this is my issue opposed to yours. Thanks for the dump though - I'll rerun the MD5 and row compare over the next couple of days with these.

Might be worth ditching this PR and starting afresh.

File: definitions/GarrFollItemSet.dbd [hunk 4]
```diff
@@ -1,5 +1,5 @@
 COLUMNS
-uint field_458934399?
+int ID
```
Comment (bloerwald) [inline definitions/GarrFollItemSet.dbd]: Nitpicky unknown here as well

File: definitions/GarrGivenName.dbd [hunk 5]
```diff
@@ -1,9 +1,9 @@
 COLUMNS
-uint field_2710791686?
 string field_978907121?
 uint field_3259647487?
+int ID
 
 BUILD 6.0.1.18179
-field_2710791686<32>
+$id$ID<32>
```
Comment (bloerwald) [inline definitions/GarrGivenName.dbd]: And here. I’d either have them all as guesses or none. Your and martin’s vote which direction.

File: definitions/ItemModifiedAppearance.dbd [hunk 6]
```diff
@@ -1,10 +1,9 @@
 COLUMNS
-uint field_3565994245?
 uint field_2556328986?
 uint field_2595378131?
 uint field_2799832687?
 uint field_516438342?
-uint field_3478340450?
+uint field_3478340450
```
Comment (bloerwald) [inline definitions/ItemModifiedAppearance.dbd]: Privably unintended.

**Comment by bloerwald** [comment]
On “unverified ID columns”: either all `?` or none. I don’t care which way around.

**Comment by barncastle** [comment]
- Added in Ids where there was `field_*<32>` or `padding_*<8>[4]` as the first column and where Ids were applicable i.e. used as a key in a hashmap in the client, values were unique and looked (yeah scientific I know) like the correct range and/or matched other build's known Id ranges
- Switched all Ids to unverified as per @bloerwald's suggestion (might as well start off with good intentions!) with the exception of other builds already having a known Id name or if they were gametables
- Fixed the above noted problems
- Merged matching entries together

**Pull request merged**